File: Core/src/main/java/com/inverce/mod/core/threadpool/DefaultHandlerThread.java
package com.inverce.mod.core.threadpool;

import android.os.Handler;
import android.os.HandlerThread;
import android.os.Looper;
import android.support.annotation.NonNull;

import java.util.concurrent.Executor;


public class DefaultHandlerThread extends HandlerThread implements Executor {
    Handler handler;
    public DefaultHandlerThread() {
        super("Handler-thread", Thread.NORM_PRIORITY);
    }

    public Handler getHandler() {
        if (handler == null) {
            Looper looper = getLooper();
            if (looper != null) {
                handler = new Handler(getLooper());
            }
        }
        if (handler == null) {
            throw new IllegalStateException("Looper not prepared, or thread is no longer active");
        }
        return handler;
    }

    @Override
    public void execute(@NonNull Runnable command) {
        getHandler().post(command);
    }
}

File: Core/src/main/java/com/inverce/mod/v2/core/threadpool/DefaultHandlerThread.kt
package com.inverce.mod.v2.core.threadpool

import android.os.Handler
import android.os.HandlerThread
import java.util.concurrent.Executor

open class DefaultHandlerThread : HandlerThread("Handler-thread", Thread.NORM_PRIORITY), Executor {
    protected val mHandler by lazy { Handler(looper) }

    override fun execute(command: Runnable) {
        mHandler.post(command)
    }
}